v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsSimpleIT.java

```diff
@@ -0,0 +1,242 @@
+    // 2. Write a single record to Spanner
+    // 3. Wait on GCS to have the file
+    // 4. Match the PK in GCS with the PK written to Spanner
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    SpannerRowsCheck.builder(
+                            spannerMetadataResourceManager, "spanner_to_gcs_metadata")
+                        .setMinRows(1)
+                        .setMaxRows(1)
+                        .build()))
+            .build();
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(10)), conditionCheck);
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+    // Perform writes to Spanner
+    writeSpannerData();
+    // Assert file present in GCS with the needed data
+    assertFileContentsInGCS();
+  }
+
+  private void createSpannerDatabase() throws IOException {
+    spannerResourceManager =
+        SpannerResourceManager.builder("rr-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String ddl =
+        String.join(
+            " ", Resources.readLines(Resources.getResource(spannerDdl), StandardCharsets.UTF_8));
+    ddl = ddl.trim();
+    String[] ddls = ddl.split(";");
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    spannerDatabaseName = spannerResourceManager.getDatabaseId();
+  }
+
+  private void createAndUploadShardConfigToGcs() throws IOException {
+    Shard shard = new Shard();
+    shard.setLogicalShardId("testA");
+    shard.setUser("dummy");
+    shard.setHost("dummy");
+    shard.setPassword("dummy");
+    shard.setPort("3306");
+    JsonObject jsObj = (JsonObject) new Gson().toJsonTree(shard).getAsJsonObject();
+    jsObj.remove("secretManagerUri"); // remove field secretManagerUri
+    JsonArray ja = new JsonArray();
+    ja.add(jsObj);
+    String shardFileContents = ja.toString();
+    LOG.info("Shard file contents: {}", shardFileContents);
+    // -DartifactBucket has the bucket name
+    gcsClient.createArtifact("input/shard.json", shardFileContents);
+  }
+
+  private void uploadSessionFileToGcs() throws IOException {
+    gcsClient.uploadArtifact(
+        "input/session.json", Resources.getResource(sessionFileResourceName).getPath());
+  }
+
+  private void createSpannerMetadataDatabase() throws IOException {
+    spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
```

Comment:
Is this needed?

Reply:
Yeah, the database does not get created unless there is a DDL.